### ISSUE TITLE
`user@0.service` failed

### ISSUE THREAD

**Issue by Xenfo** [arkane-systems/bottle-imp#21]
This error only happens after WSL has been running for a while, when I restart WSL everything works fine. Might have something to do with running WSL from admin terminal.
```
imp: WARNING: systemd is in degraded state, issues may occur!
❯ systemctl --failed
  UNIT           LOAD   ACTIVE SUB    DESCRIPTION
● user@0.service loaded failed failed User Manager for UID 0

LOAD   = Reflects whether the unit definition was properly loaded.
ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
SUB    = The low-level unit activation state, values depend on unit type.
1 loaded units listed.
❯ systemctl status user@0.service
× user@0.service - User Manager for UID 0
     Loaded: loaded (/lib/systemd/system/user@.service; static)
    Drop-In: /usr/lib/systemd/system/user@.service.d
             └─timeout.conf
     Active: failed (Result: exit-code) since Sat 2022-10-15 10:21:29 EDT; 25min ago
       Docs: man:user@.service(5)
    Process: 795 ExecStart=/lib/systemd/systemd --user (code=exited, status=1/FAILURE)
   Main PID: 795 (code=exited, status=1/FAILURE)

Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[1]: Starting User Manager for UID 0...
Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[795]: pam_unix(systemd-user:session): session opened for user root(uid=0) by (u>
Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[795]: pam_systemd(systemd-user:session): Runtime directory '/run/user/0' is not>
Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[795]: pam_systemd(systemd-user:session): Not setting $XDG_RUNTIME_DIR, as the d>
Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[795]: Trying to run as user instance, but $XDG_RUNTIME_DIR is not set.
Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[1]: user@0.service: Main process exited, code=exited, status=1/FAILURE
Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[1]: user@0.service: Failed with result 'exit-code'.
Oct 15 10:21:29 DESKTOP-RVHOL00 systemd[1]: Failed to start User Manager for UID 0.
lines 1-17/17 (END)
```

WSL info:
```
WSL version: 0.70.0.0
Kernel version: 5.15.68.1
WSLg version: 1.0.45
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.674
```

**Comment by cerebrate** [comment]
I think that one's a result of running WSL as root, rather than as Windows administrator, combined with what appears to be an issue in WSL's own systemd support rather than bottle-imp. I'll investigate further.

**Comment by cerebrate** [comment]
Yep, this is a problem with WSL automatically bind-mounting `/mnt/wslg` over the user runtime directory, which kicks up this problem for every user except the one with uid 1000. Reported over there as microsoft/WSL#9025 .

Leaving this open for now while I contemplate potential workarounds for bottle-imp 0.9.

**Comment by cerebrate** [comment]
Fixed in 0.9.0, now available:

https://github.com/arkane-systems/bottle-imp/releases/tag/v0.9.0
